style.css

```diff
@@ -1,19 +1,84 @@
+:root {
+	--primary-background-color: hsl(24 20% 95%);
+	--primary-foreground-color: hsl(24 20% 5%);
+	--secondary-background-color: hsl(200 10% 20%);
+	--secondary-foreground-color: white;
+	--tertiary-background-color: transparent;
+	--tertiary-foreground-color: hsl(200 10% 20%);
+	--primary-shadow-color: white;
+	--secondary-shadow-color: hsl(0 0% 0% / .5);
+	--primary-border-color: hsl(0 0% 0% / .25);
+	--secondary-border-color: var(--secondary-background-color);
+	--shade-color: hsl(0 0% 0% / .3);
+
+	--spec-link-background-color: var(--secondary-background-color);
+	--spec-link-foreground-color: var(--secondary-foreground-color);
+	--highlight-background-color: #f06;
+	--highlight-foreground-color: white;
+	--prefix-background-color: var(--shade-color);
+	--prefix-foreground-color: var(--spec-link-foreground-color);
+	--progress-color: var(--shade-color);
+
+	--pass-color: #490;
+	--almost-pass-color: #8c0;
+	--slightly-buggy-color: #bb0;
+	--buggy-color: #e80;
+	--very-buggy-color: #e40;
+	--fail-color: #e20;
+	--epic-fail-color: #b00;
+
+	--aside-border-color: hsl(0 0% 80% / .8);
+	--aside-background-color: hsl(0 0% 98% / .6);
+	--aside-foreground-color: black;
+}
+
+@media (prefers-color-scheme: dark) {
+	:root {
+		--primary-background-color: hsl(24 20% 5%);
+		--primary-foreground-color: hsl(24 20% 95%);
+		--secondary-background-color: hsl(200 10% 20%);
+		--secondary-foreground-color: white;
+		--tertiary-background-color: transparent;
+		--tertiary-foreground-color: hsl(200 10% 90%);
+		--primary-shadow-color: black;
+		--secondary-shadow-color: hsl(0 0% 25% / .5);
+		--primary-border-color: hsl(0 0% 25% / .25);
```

Comment:
A lot of these are identical between the modes, or only differ in lightness. If that's actually the case, what about this approach? https://lea.verou.me/2021/03/inverted-lightness-variables/

Reply:
While I can see your point in just changing the luminance for the dark mode, I have to say that I still prefer writing out the colors.
The approach you outline in the blog post has several downsides, in my opinion:

1. You can't see and change the colors in developer tools (neither the browser tools nor in IDEs like VSCode).
2. You lose the flexibility to create custom themes or even slight differences. To allow different combinations of hs and l you still need to define separate custom properties.
3. The custom property names may be unclear and even misleading to some authors. It isn't obvious that `l` stands for luminance and `--l-0` seems to indicate "no luminance".

Reply:
For 2, you can still have color variables where actually needed. It's not like you're bound to only vary lightness, forever. If you actually need to override an entire color, you can override an entire color. Also, we can still use color variables *as well*, for clarity, we just wouldn't need to override their values in dark mode.
For 3, using `--l-*` is not an integral part of the technique, you can use `--lightness-*` instead (it's not luminance — that's another thing).

Reply:
I am still not convinced that simpler changing of the lightness justifies the longer and more complex final value creation, though I did it for you. Please let me know if that's what you had in mind!

And thanks for the hint that lightness is not luminance! I have to admit, my knowledge about color spaces and related terminology is pretty limited.